### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by GiuseppeTG**
### Project requirements
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<main>
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb:1]: - [ ] @Team great work on your app, however, I was unable to move past the login or registration page, I successfully submit my details but it doesn't log me in kindly take  a look at it, here is a snapshot of users created using the form 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @GiuseppeTG and @waqaskanju , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (GiuseppeTG) [reply app/views/devise/registrations/new.html.erb:1]: Hello @danmainah. It works just fine to me

I don't know what can be the problem in this case 😅 

https://user-images.githubusercontent.com/86022690/192836952-da7c2af4-e6f9-4373-a0a5-f776a882bffc.mov

Comment (GiuseppeTG) [reply app/views/devise/registrations/new.html.erb:1]: @danmainah  Sorry for the video quality but it was for making it less than 10 MB

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @GiuseppeTG, 

###  For now your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nicupop729 — COMMENTED** [review]
Hello @GiuseppeTG 👋,

_This is an additional review as a result of the second opinion request._

I am glad to see that you conquer your blocker and your project was approved!
Kudos for that! :clap:

Please be aware that we are not allowed to use the same branch and the same PR for multiple milestones. 

You used the same branch for `Blog app: add devise` and for `Blog app: add authorization rules`.
Next time, a diligent code reviewer will invalidate your PR.

Please always use different branches/PRs for each milestone.

Good luck and keep rocking 🚀!

**Pull request merged**